### PR TITLE
Added 2007 Q2

### PR DESCRIPTION
**Pull request by macaquedev**

### CODE REVIEW

File: solutions/bio1/MuTorere/index.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                exactly what is asked for. We can see that to solve the
+                $n$ command, we can just make a move exactly as
+                described in the problem statement, but it is not
+                immediately clear how to solve the $r$ command.
+            </SText>
+            <STitle>Solution</STitle>
+            <SSubtitle>Dealing with the $r$ command</SSubtitle>
+            <SText>
+                Notice that the game only has $9$ circles - the
+                "putahi" and the $8$ "kawai". Each circle can be in one of $3$ states:
```
Comment (AG-88301) [inline solutions/bio1/MuTorere/index.tsx:19]: putahi and kawai better in italics i think

**Comment by AG-88301** [comment]
Other than that I think it's fine... nice editorial and clean code

**Review by AG-88301 — CHANGES_REQUESTED** [review]
kawai and putahi in italics

File: solutions/bio1/MuTorere/index.tsx (hunk 1 above)
Comment (macaquedev) [reply solutions/bio1/MuTorere/index.tsx:19]: How do you do italics?

File: solutions/bio1/MuTorere/index.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            <SSubtitle>Representing the game states</SSubtitle>
+            <SText>
+                In implementation questions like this, it is often useful to
+                use object-oriented programming. We define a class <SCode>Board</SCode>,
+                which represents a game state, and it has two attributes:
+                <SCode>chars</SCode> and <SCode>current_turn</SCode>. The variable
+                <SCode>chars</SCode> is a string of length $9$, which stores the
+                board state in the form of the input, and <SCode>current_turn</SCode>
+                is an integer which stores whose turn it is. A neat trick when
+                <SCode>current_turn</SCode> is either $1$ or $2$ is to use
```
Comment (AG-88301) [inline solutions/bio1/MuTorere/index.tsx]: missing a space after use

File: solutions/bio1/MuTorere/index.tsx (hunk 1 above)
Comment (AG-88301) [reply solutions/bio1/MuTorere/index.tsx]: <i>text</i>
```
<i>text</i>
```

File: solutions/bio1/MuTorere/index.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                cannot simply add them to a set. This is because the default
+                implementation of the <SCode>__hash__</SCode> function for classes
+                is to return the memory address of the object, which means that 
+                two different <SCode>Board</SCode> class instances representing 
+                the same game state will have different hash values, and therefore 
+                will not be considered equal when added to a set. To fix this, instead 
+                of directly adding the <SCode>Board</SCode> instance to the set, we
+                can make a new string, which is the concatenation of the 
+                two attributes of the <SCode>Board</SCode> instance, and add that
+                to the set instead. This guarantees that two different <SCode>Board</SCode>
```
Comment (AG-88301) [inline solutions/bio1/MuTorere/index.tsx]: missing a space after board

File: solutions/bio1/MuTorere/index.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                the game until either a previous game state is reached, or the game
+                ends. If a previous game state is reached, it is clear that the result
+                will be a draw, because an infinite loop has been reached.
+            </SText>
+            <SSubtitle>Representing the game states</SSubtitle>
+            <SText>
+                In implementation questions like this, it is often useful to
+                use object-oriented programming. We define a class <SCode>Board</SCode>,
+                which represents a game state, and it has two attributes:
+                <SCode>chars</SCode> and <SCode>current_turn</SCode>. The variable
```
Comment (AG-88301) [inline solutions/bio1/MuTorere/index.tsx]: missing a space after variable

File: solutions/bio1/MuTorere/index.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                <SCode>chars</SCode> is a string of length $9$, which stores the
+                board state in the form of the input, and <SCode>current_turn</SCode>
```
Comment (AG-88301) [inline solutions/bio1/MuTorere/index.tsx]: missing space end of line

**Review by AG-88301 — APPROVED** [review]
good

**Pull request merged**